packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts

```diff
@@ -112,6 +116,9 @@ export class AzureStorageDeployDriverImpl extends AzureDeployImpl {
     );
     const responses = await Promise.all(tasks);
     const errorResponse = responses.find((res) => res.errorCode);
+    if (errorResponse?._response?.status === HttpStatusCode.INTERNAL_SERVER_ERROR) {
+      throw DeployExternalApiCallError.uploadToStorageRemoteError(sourceFolder, errorResponse);
+    }
```

Comment:
Nit: To make handling of `errorResponse` in one section, one way is:
`
if (errorResponse) {
  if (errorResponse._response?.status === 500) throw error.
  else throw someOtherError.
}
}
`